pkg/kubectl/cmd/login.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+)
+
+func NewCmdLogin(f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "login Log in to the cluster.",
+		Long:  `login obtains the neccesary credentials to make authenticated requests to the cluster. The type of log in depends on the 'auth-provider' type in your configuration, and will likely require some type of user interaction.`,
```

Comment:
Isn't that case is supported by `--user=XXX` ? An `AuthProvider` belongs to a `AuthInfo` (i.e. within `user` in the `.kube/config`), they don't exist on their own in in the Config.

As for an initial bootstrapping flag, what did you have in mind there? Each AuthProvider has its own configuration, so maybe a flag which takes a bunch of key/value properties separated by some delimiter?